client/src/components/feedback/edit-question-for-queue-modal.tsx

```diff
@@ -0,0 +1,275 @@
+/*
+This software is Copyright ©️ 2020 The University of Southern California. All Rights Reserved. 
+Permission to use, copy, modify, and distribute this software and its documentation for educational, research and non-profit purposes, without fee, and without a written agreement is hereby granted, provided that the above copyright notice and subject to the full license file found in the root of this software deliverable. Permission to make commercial use of this software may be obtained by contacting:  USC Stevens Center for Innovation University of Southern California 1150 S. Olive Street, Suite 2300, Los Angeles, CA 90115, USA Email: accounting@stevens.usc.edu
+
+The full terms of this copyright and license should always be found in the root directory of this software deliverable as "license.txt" and if these terms are not found with this software, please contact the USC Stevens Center for the full license.
+*/
+import {
+  Backdrop,
+  Button,
+  Fade,
+  FormControl,
+  Grid,
+  InputLabel,
+  makeStyles,
+  Modal,
+  Select,
+  TextField,
+  Theme,
+  Typography,
+} from "@material-ui/core";
+import React, { useState } from "react";
+import {
+  Category,
+  Mentor,
+  Question,
+  QuestionType,
+  Subject,
+  Topic,
+  UtteranceName,
+} from "types";
+import { onTextInputChanged } from "helpers";
+import { v4 as uuid } from "uuid";
+import { Autocomplete } from "@material-ui/lab";
+import { addOrUpdateSubjectQuestions, addQuestionToRecordQueue } from "api";
+import { SubjectQuestionGQL } from "types-gql";
+
+const useStyles = makeStyles((theme: Theme) => ({
+  homeThumbnail: {
+    position: "relative",
+    width: "78%",
+    height: 180,
+  },
+  siteThumbnail: {
+    width: 180,
+    height: 135,
+  },
+  title: {
+    fontWeight: "bold",
+  },
+  inputField: {
+    width: "100%",
+    margin: 10,
+  },
+  modal: {
+    display: "flex",
+    alignItems: "center",
+    justifyContent: "center",
+  },
+  paper: {
+    backgroundColor: theme.palette.background.paper,
+    border: "2px solid #000",
+    boxShadow: theme.shadows[5],
+    padding: theme.spacing(2, 4, 3),
+    maxWidth: "50%",
+  },
+}));
+
+function EditQuestionForQueueModal(props: {
+  handleClose: () => void;
+  open: boolean;
+  mentor: Mentor;
+  userQuestion: string;
+  accessToken: string;
+}): JSX.Element {
+  const { handleClose, open, userQuestion, mentor, accessToken } = props;
+  const classes = useStyles();
+  const [selectedSubject, setSelectedSubject] = useState<Subject>();
+  const [selectedCategory, setSelectedCategory] = useState<Category>();
+  const [customQuestion, setCustomQuestion] = useState<string>(userQuestion);
+  const selectedTopics: Topic[] = [];
```

Comment:
I'd be wary of storing data that changes as a class variable. Changes to class variables do not trigger re-renders of the page. So, if anything relies on this variable to display anything, this has to be a state variable. Else, it should be fine.